### ISSUE TITLE
False Positive unsecure_fae

### ISSUE THREAD

**Issue by OberstVonGatow** [abaplint/abaplint#2384]
I get lint errors on unsecure_fae.
Even if i try in the playground i get the same error.
Or do i missunderstand the rule?

`DATA lt_data TYPE STANDARD TABLE OF mara.

IF lt_data IS NOT INITIAL.
  SELECT FROM mara
    FIELDS matnr
    FOR ALL ENTRIES IN @lt_data
    WHERE matnr = @lt_data-matnr
    INTO TABLE @DATA(lt_mara)
    ORDER BY matnr.
  IF sy-subrc = 0.
    WRITE lt_mara[ 1 ]-matnr.
  ENDIF.
ENDIF.`

**Comment by larshp** [comment]
thanks, its currently marked as "experimental", the small icon. Hmm, perhaps I should make it better vissible.

It will eventually work, hopefully :) anyhow another test case to add

![image](https://user-images.githubusercontent.com/5888506/152120543-462d9fcc-def9-4aaa-b590-38defdd9931d.png)

**Comment by OberstVonGatow** [comment]
Ah okay sorry i didn't realized it was experimental. Should i close the Issue or keep it open as reminder?

**Comment by larshp** [comment]
keep it open please, I'll add this scenario as a unit test, when I get back to fixing/implementing the rule

**Comment by FreHu** [comment]
one more variant with `check`
```abap
    CHECK e071_data IS NOT INITIAL.
    SELECT * FROM e070 FOR ALL ENTRIES IN @e071_data WHERE trkorr     = @e071_data-trkorr AND
                                                         ( trfunction = 'K' OR
                                                           trfunction = 'W' OR
                                                           trfunction = 'S' )
                                                     INTO CORRESPONDING FIELDS OF TABLE @data. "#EC CI_SUBRC
```

**Comment by larshp** [comment]
its a bug

ABAP is a crazy language, if more people support the project, then more bugs will be fixed

**Comment by larshp** [comment]
also note the experimental flag,

![image](https://user-images.githubusercontent.com/5888506/233299048-0cd5fd14-6f6b-40d2-b39d-2320f6f490b8.png)

**Comment by Zeatynis** [comment]
> Is the `ORDER BY` addition now allowed with `FOR ALL ENTRIES`? I am on v750 and it is not throwing any exception in VSCode w/ abaplint, but I can't use it in my system

Read manual. It is allowed in some cases.
![obraz](https://user-images.githubusercontent.com/9119187/233327458-181361df-84a0-4cf8-963e-553ac9e87081.png)

**Comment by Zeatynis** [comment]
> > > Is the `ORDER BY` addition now allowed with `FOR ALL ENTRIES`? I am on v750 and it is not throwing any exception in VSCode w/ abaplint, but I can't use it in my system
> > 
> > 
> > Read manual. It is allowed in some cases. ![obraz](https://user-images.githubusercontent.com/9119187/233327458-181361df-84a0-4cf8-963e-553ac9e87081.png)
> 
> Which is not what's happening in the opening comment, as there's no `PRIMARY KEY` addition. So my question is: regardless of how the unsecure_fae is handled today by abaplint, would the top comment syntax compile at all in an actual system? Because on ABAP 750, it throws a syntax error: ![image](https://user-images.githubusercontent.com/104131979/233331616-848fe57d-7570-4ea6-a93e-43f9e830a400.png)

Move order by before into table.
